genie-agent/src/main/java/com/netflix/genie/agent/execution/services/AgentJobService.java

```diff
@@ -39,6 +40,16 @@
 @Validated
 public interface AgentJobService {
 
+    /**
+     * Perform server handshake. Before going any further, ensure the server is reachable and that this agent is
+     * compatible with it.
+     *
+     * @param agentClientMetadata metadata about the client making this request
+     * @throws HandshakeException if the server rejects this client
+     */
+    void handshake(
```

Comment:
I agree in principle, but if you look at the rest of this interface, it's the pattern we've been using.
For example, failure to reserve a job, resolve a specification, etc. could all return some sort of "outcome object" with an details.
What we're doing instead is either returning the result of successful operation or an exception with details.

We could deviate from this for this specific `handshake` API, I thought about it.
I just saw more value in sticking to the existing convention, since the alternative (outcome return value) doesn't add a lot of value.